File: src/main/java/pro/sky/java/weatherfrontend/handler/WeatherExceptionHandler.java
package pro.sky.java.weatherfrontend.handler;

import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import pro.sky.java.weatherfrontend.exception.CityNotFoundException;

@ControllerAdvice
public class WeatherExceptionHandler {

    @ExceptionHandler(CityNotFoundException.class)
    public String cityNotFoundException(CityNotFoundException e, Model model) {
        model.addAttribute("city", e.getCity());
        e.printStackTrace();
        return "error";
    }
}

File: src/main/kotlin/pro/sky/java/weatherfrontend/handler/WeatherExceptionHandler.kt
package pro.sky.java.weatherfrontend.handler

import org.springframework.ui.Model
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import pro.sky.java.weatherfrontend.exception.CityNotFoundException
import pro.sky.java.weatherfrontend.html.errorPage

@RestControllerAdvice
class WeatherExceptionHandler {
    @ExceptionHandler(CityNotFoundException::class)
    fun cityNotFoundException(e: CityNotFoundException, model: Model): String {
        return errorPage(e.city).also {
            e.printStackTrace()
        }
    }
}